### PR TITLE
pPacketData free bug resolved

### PR DESCRIPTION
**Pull request by vtpl1**
@rarzumanyan,
Putting `pPacketData = Buffer::MakeOwnMem(sizeof(PacketData));` instead of `pPacketData = Buffer::MakeOwnMem(1U);` solved the issue.

I am not sure why it was like `pPacketData = Buffer::MakeOwnMem(1U);` in the first place.
This addresses the issue #204 
BR,
Monotosh

### CODE REVIEW

**Review by rarzumanyan — CHANGES_REQUESTED** [review]
Hi @vtpl1 
Thanks for the PR!

Could you please clean up the patch? `Buffer::MakeOwnMem()` fix makes total sense, but other changes are IMO irrelevant.

File: PyNvCodec/TC/inc/MemoryInterfaces.hpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ class DllExport Buffer final : public Token {
   bool Allocate();
   void Deallocate();
 
-  bool own_memory = true;
```
[the remark this answers is not in the document]
Comment (vtpl1) [reply PyNvCodec/TC/inc/MemoryInterfaces.hpp:71]: By default no memory is associated so why to own a NULL memory?
Though it does not solve the bug.

[the remark this answers is not in the document]
Comment (rarzumanyan) [reply PyNvCodec/TC/inc/MemoryInterfaces.hpp:71]: @vtpl1 

Yes, you're right about that - by default `Buffer` doesn't have any memory allocated so changing the default value won't affect the behavior.

In order to minimize the amount of changes to the code it's kinda "usual practice" to change code only if it's required to fix something or add new feature. Following this approach a person who sees your patch with "fixing pPacketData bug"-like commit message will have a good insight on what does the patch do. They would not probably expect this to introduce changes to one of the core library headers, right?

**Comment by vtpl1** [comment]
@rarzumanyan ,
Incorporated the changes,
But I am not sure why MuxFrame removal is showing as my commit.

**Comment by rarzumanyan** [comment]
@vtpl1 

Unfortunately the PR looks smashed and it contain previous patches as well (7 files modified overall). 
That could probably happen because of force push.
Could you just create a new PR with only ` PyNvCodec/TC/src/Tasks.cpp` file modified?

**Comment by vtpl1** [comment]
Yes, that is what I feel, can I include the copyright notice for other files for older PR s?

**Comment by rarzumanyan** [comment]
> 
> 
> @rarzumanyan
> Yes, that is what I feel, can I include the copyright notice for other files for older PR s?

I assume we can do as follows: please add your company name to the file which is modified by your latest PR and meanwhile I will consult with our legals if copyright can be added to the files affected by your previous contributions. 

Does this approach make sense to you?

**Comment by vtpl1** [comment]
ok sournds good

**Pull request closed without merging**